File: GOF/Bridge/cpp/WindowSystemFactory.cpp
//
// Created by 17271 on 2019/3/30.
//

#include "WindowSystemFactory.h"
#include "Window/WindowImp/XWindowImp.h"

using std::make_shared;

WindowSystemFactory& WindowSystemFactory::getInstance()
{
	static WindowSystemFactory factory;
	return factory;
}

std::shared_ptr<WindowImp> WindowSystemFactory::makeWindowImp()
{
	return make_shared<XWindowImp>();
}

File: GOF/Bridge/java/src/Factory/WindowSystemFactory.java
package Factory;

import Window.WindowImp.WindowImp;
import Window.WindowImp.XWindowImp;

public class WindowSystemFactory {

	private static final WindowSystemFactory factory = new WindowSystemFactory();

	public static WindowSystemFactory getInstance() {
		return factory;
	}

	public WindowImp makeWindowImp() {
		return new XWindowImp();
	}
}
